test/Integration.t.sol

```diff
@@ -0,0 +1,311 @@
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_250);
+
+        // Check correct cap increase after cooldown
+        skip(24 hours);
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_500);
+        // initialSupply + suppliedBefore + newlySupplied + gap = 750 + 2_000 + 250 + 500 = 3_500
+
+        vm.roll(block.number + 1);
+        vm.prank(user);
+        pool.withdraw(WBTC, 125e8, user);
+
+        // Check correct cap decrease
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_375);
+        // previousSupply + gap - justWithdrawn = 3_000 + 500 - 125 = 3_375
+
+        vm.prank(user);
+        pool.withdraw(WBTC, 125e8, user);
+
+        // Check the cap is not changing in the same block
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_375);
+
+        // Check correct cap decrease after block changes
+        vm.roll(block.number + 1);
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_250);
+        // previousSupply + gap - justWithdrawn = 2_875 + 500 - 125 = 3_250
+
+        assertEq(ERC20(WETH).decimals(), 18);
+
+        DataTypes.ReserveData memory wethReserveData = pool.getReserveData(WETH);
+
+        // Confirm initial borrow cap
+        uint256 initialBorrowCap = wethReserveData.configuration.getBorrowCap();
+        assertEq(initialBorrowCap, 1_400_000);
+
+        // Confirm initial borrows
+        uint256 initialBorrows = ERC20(wethReserveData.variableDebtTokenAddress).totalSupply() / 10 ** ERC20(wethReserveData.variableDebtTokenAddress).decimals();
+        assertEq(initialBorrows, 126_530);
+
+        vm.prank(SPARK_PROXY);
+        capAutomator.setBorrowCapConfig({
+            asset:            WETH,
+            max:              2_000_000,
+            gap:              100_000,
+            increaseCooldown: 12 hours
+        });
+
+        // Check correct cap decrease
+        capAutomator.execBorrow(WETH);
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 226_530);
+        // totalDebt + gap = 126_530 + 100_000 = 226_530
+
+        vm.prank(user);
+        pool.borrow(WETH, 470e18, 2 /* variable rate mode */, 0, user);
+
+        // Check that another cap change is not possible in the same block
+        capAutomator.execBorrow(WETH);
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 226_530);
+
+        // Check correct cap increase in the new block
+        vm.roll(block.number + 1);
+        capAutomator.execBorrow(WETH);
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 227_000);
+        // totalDebt + gap = initialBorrows + newlyBorrowed + gap = 126_530 + 470 + 100_000 = 227_000
```

Comment:
I would put this into a separate test and make the names of the test more specific.